src/Microsoft.AspNetCore.StaticFiles/StaticFileContext.cs

```diff
@@ -308,11 +309,18 @@ public void ApplyResponseHeaders(int statusCode)
                 // it is not returned for 304, 412, and 416
                 _response.ContentLength = _length;
             }
-            _options.OnPrepareResponse(new StaticFileResponseContext()
+            var staticFileContext = new StaticFileResponseContext
             {
                 Context = _context,
                 File = _fileInfo,
-            });
+            };
+
+            var cacheProfile = _options.CacheProfileProvider(staticFileContext) ?? _options.CacheProfile;
```

Comment:
`_options?.CacheProfileProvider` or check that CacheProfileProvider != null in the middleware constructor